python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -372,12 +468,127 @@ def predict_proba(self, dataset, iou_threshold=0.1):
-        type iou_threshold: float
+        :type iou_threshold: float
         """
         predictions = self.predict(dataset, iou_threshold)
-        prob_scores = [[pred.class_scores for pred in image_prediction] for image_prediction in predictions]
+        prob_scores = [[pred.class_scores for pred in image_prediction]
+                       for image_prediction in predictions]
+        return prob_scores
+
+
+class WrappedMlflowAutomlObjectDetectionModel:
+    """A class for wrapping an AutoML for images MLflow object detection model
+        in the scikit-learn style."""
+
+    def __init__(self, model: PyFuncModel,
+                 classes: Union[list, np.array]) -> None:
+        """Initialize the WrappedMlflowAutomlObjectDetectionModel.
+
+        :param model: mlflow model
+        :type model: mlflow.pyfunc.PyFuncModel
+        :param classes: list of class names
+        :type classes: list or np.array
+        """
+
+        self._model = model
+        self._classes = classes
+        self._label_dict = {label: (i+1) for i, label in enumerate(classes)}
+
+    def _mlflow_predict(self, dataset: pd.DataFrame) -> pd.DataFrame:
+        """Perform the inference using the wrapped MLflow model.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: pandas.DataFrame
+        :return: The predicted data.
+        :rtype: pandas.DataFrame
+        """
+
+        predictions = self._model.predict(dataset)
+        return predictions
+
+    def predict(self, dataset: pd.DataFrame, iou_thresh: float = 0.5,
+                score_thresh: float = 0.5):
+        """Create a list of detection records from the image predictions.
+
+        :param dataset: The dataset to predict on.
+        :type dataset: pandas.DataFrame
+        :param iou_thresh: Intersection-over-Union (IoU) threshold for NMS (or
+                           the amount of acceptable error). Objects with error
+                           cores higher than the threshold will be removed.
+        :type iou_thresh: float
+        :param score_thresh: Threshold to filter detections based on
+                            predicted confidence scores.
+        :type score_thresh: float
+        :return: Final detections from the object detector
+        :rtype: List of Detection Records
+
+
+        Below is example Label (y) representation for a cohort of 2 images,
+        with 3 objects detected for the first image and 1 for the second image.
+
+        [
+         [
+            [class, topX, topY, bottomX, bottomY, (optional) confidence_score],
+            [class, topX, topY, bottomX, bottomY, (optional) confidence_score],
+            [class, topX, topY, bottomX, bottomY, (optional) confidence_score],
+         ],
+         [
+            [class, topX, topY, bottomX, bottomY, (optional) confidence_score],
+         ]
+
+        ]
+        """
+        image_sizes = dataset['image_size']
```

Comment:
it seems this is not tested in code coverage:
https://app.codecov.io/gh/microsoft/ml-wrappers/commit/75da425c53d2feb128f7301e464fb3d31564e9d4/blob/python/ml_wrappers/model/image_model_wrapper.py

can you add a test for this method?